### PR TITLE
Refactor gardener-seed-admission-controller deployment into component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/priority normal
/topology seed
/platform all
/exp intermediate

**What this PR does / why we need it**:
This PR refactors the deployment for the gardener-seed-admission-controller and adopts the component concept. Consequently, it eliminates the Helm chart and adds full unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+		v1beta1constants.LabelApp:  "gardener",
+		v1beta1constants.LabelRole: "seed-admission-controller",
+	}
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (g *gardenerSeedAdmissionController) Wait(ctx context.Context) error {
```
Comment (ialidzhikov) [inline pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go:371]: You could add doc string.

File: pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilManagedResourceHealthy(timeoutCtx, g.client, g.namespace, managedResourceName)
+}
+
+func (g *gardenerSeedAdmissionController) WaitCleanup(ctx context.Context) error {
```
Comment (ialidzhikov) [inline pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go:378]: You could add doc string.

**Review by ialidzhikov — COMMENTED** [review]
/lgtm
/needs second-opinion

File: pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go:371]: We don't add doc-strings for struct methods where the struct is private, or?

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**